File: serviceAccount/src/main/sdk/com/effective/android/service/account/AccountSdk.java
package com.effective.android.service.account;

import io.reactivex.Flowable;

public interface AccountSdk {

    void login(String returnActivityPath);

    void logout();

    void logout(boolean forceLogin);

    void logout(boolean forceLogin, String returnActivityPath);

    String getLoginPath();

    Flowable<Account> getAccount();
}

File: serviceAccount/src/main/sdk/com/effective/android/service/account/AccountSdk.kt
package com.effective.android.service.account

import android.content.Context
import io.reactivex.Flowable


interface AccountSdk {

    fun isLogin(): Flowable<Boolean>

    fun logout()

    fun login(context: Context)

    fun getAccount(): Flowable<UserInfo>

    fun addAccountChangeListener(accountChangeListener: AccountChangeListener)

    fun removeAccountChangeListener(accountChangeListener: AccountChangeListener)
}
